beacon_node/network/src/attestation_service/mod.rs

```diff
@@ -186,18 +186,50 @@ impl<T: BeaconChainTypes> AttestationService<T> {
     pub fn validator_subscriptions(
         &mut self,
         subscriptions: Vec<ValidatorSubscription>,
-    ) -> Result<(), ()> {
+    ) -> Result<(), String> {
+        let current_slot = self
+            .beacon_chain
+            .slot()
+            .map_err(|e| format!("Failed to get current slot: {:?}", e))?;
+
+        let mut state = self
```

Comment:
I think we can get rid of this guy now, it's a very expensive way to read the current slot :D

Reply:
whoops!